Sources/SignalRClient/HubConnection.swift

```diff
@@ -303,6 +309,8 @@ public class HubConnection {
 
     fileprivate func connectionDidReceiveData(data: Data) {
         logger.log(logLevel: .debug, message: "Data received")
+        resetKeepAlive()
```

Comment:
This cannot be here - we will try to send keep alive even if the connection was not successfully open. When a connection fails to open the `connectionDidClose` is not invoked (because it was not open) so the clean up will not happen. At the very least it should be between lines 333 and 334 because we know there that the connection should be open.

Reply:
I moved it.